UnityWeld/Binding/Internal/TypeResolver.cs

```diff
@@ -310,5 +366,73 @@ public static BindableMember<PropertyInfo>[] FindBindableCollectionProperties(Co
                 .Where(p => !typeof(string).IsAssignableFrom(p.Member.PropertyType))
                 .ToArray();
         }
+
+        /// <summary>
+        /// Register custom binding attribute.
+        /// This will allow to mark bindable properties in external dlls without referencing UnityWeld.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        public static void RegisterCustomBindingAttribute<T>() where T : Attribute
+        {
+            var type = typeof(T);
+            if (BindingAttributeTypes.Contains(type))
+            {
+                return;
+            }
+
+            BindingAttributeTypes.Add(type);
+        }
+
+        /// <summary>
+        /// Register custom ViewModel provider.
+        /// This will allow to use custom ViewModel providers in external dlls without referencing UnityWeld.
+        /// </summary>
+        /// <param name="provider"></param>
+        public static void RegisterCustomViewModelProvider(ViewModelProviderDelegate provider)
+        {
+            if (ViewModelProviders.Contains(provider))
+            {
+                return;
+            }
+
+            ViewModelProviders.Add(provider);
+        }
+
+        /// <summary>
+        /// Check if type has binding attribute
+        /// </summary>
+        public static bool HasBindingAttribute(this MemberInfo type)
+        {
+            //for to avoid GC

```

Comment:
I can't believe I didn't hear about this :) I work with Unity since 3.5 and used to use 'for' already. Thanks. I'll gladly change to foreach.
And I got the idea about initialization phase.